File: bank-account.cpp
#include <iostream>
using namespace std;
class Bank
{
private:
    string name;
    int accno;
    string acctype;
    int balance;

public:
    void assign()
    {
        cout << "Enter Name, Account number, Account type, Balance:";
        cin >> name >> accno >> acctype >> balance;
    }
    void deposit(int val)
    {
        balance += val;
    }
    void showbal()
    {
        cout<<"Balance is:"<<balance<<endl;
    }
    void withdraw(int val)
    {
        const int min = 3000;
        if ((balance - val) <= 3000)
        {
            cout << "You can withdraw only " << (balance - 3000) << endl;
        }
        else
        {
            balance -= val;
            cout << "New Balance is:" << balance << endl;
        }
    }
    void showdetail()
    {
        cout<<"Name is:"<<name<<" Balance is:"<<balance;
    }
};
int main()
{
    int val;
    Bank b;
    b.assign();
    cout<<"What amount you would like to deposit? ";
    cin>>val;
    b.deposit(val);
    b.showbal();
    cout<<"What amount you want to withdraw? ";
    cin>>val;
    b.withdraw(val);
    b.showdetail();
    cout<<endl;
    return 0;
}

File: BankAccount.java
/* Design a class to represent bank account details as base class and bank process as subclass. 
Then create bank process class which is the subclass of bank account class which contains amount 
as instance variables and methods are deposit() and withdraw() and then return balance.*/




import java.util.*;



class BankAcc {
	String name;
	int accNo;
	String accType;
	double balance;

	void getDetails() {
		Scanner sc = new Scanner(System.in);
		System.out.print("Enter the name: ");
		name = sc.nextLine();
		System.out.print("Enter the account number: ");
		accNo = sc.nextInt();
		System.out.print("Enter the account type: ");
		accType = sc.next();
		System.out.print("Enter the balance: ");
		balance = sc.nextDouble();
        
	}

	void printDetails() {
		System.out.println("Name: " + name);
		System.out.println("Account Number: " + accNo);
		System.out.println("Account Type: " + accType);
		System.out.println("Balance: " + balance);
	}
}

class BankProcess extends BankAcc {
	double amount;

	void deposit() {
		Scanner sc = new Scanner(System.in);
		System.out.print("Enter the amount to be deposited: ");
		amount = sc.nextDouble();
		balance += amount;
        
	}

	void withdraw() {
		Scanner sc = new Scanner(System.in);
		System.out.print("Enter the amount to be withdrawn: ");
		amount = sc.nextDouble();
		balance -= amount;
        
	}

	void returnBalance() {
		System.out.println("Balance: " + balance);
	}
}

public class BankAccount {
	public static void main(String[] args) {
		BankProcess p1 = new BankProcess();
		p1.getDetails();
		p1.printDetails();
		p1.deposit();
		p1.withdraw();
		p1.returnBalance();
	}
}